multicluster/apis/multicluster/v1alpha1/clusterset_types.go

```diff
@@ -21,31 +21,49 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// MemberCluster defines member cluster information.
-type MemberCluster struct {
-	// Identify member cluster in ClusterSet.
+// LeaderClusterInfo specifies information of a leader cluster.
+type LeaderClusterInfo struct {
+	// Identify a leader cluster in the ClusterSet.
 	ClusterID string `json:"clusterID,omitempty"`
-	// API server of the destination cluster.
+	// API server endpoint of the leader cluster. Used only for a member cluster.
+	// e.g. "https://172.18.0.1:6443", "https://example.com:6443".
 	Server string `json:"server,omitempty"`
-	// Secret name to access API server of the member from the leader cluster.
+	// Name of the Secret resource in the member cluster, which stores
+	// the token to access the leader cluster's API server.
 	Secret string `json:"secret,omitempty"`
-	// ServiceAccount used by the member cluster to access into leader cluster.
+	// ServiceAccount in the leader cluster, from which the member cluster's token
+	// is generated. This is an optional field which helps admin to check
+	// which ServiceAccount is used by a member cluster to access the leader cluster.
+	//
+	// DEPRECATED
+	// This field is planned to be removed in the future releases.
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 }
 
 // ClusterSetSpec defines the desired state of ClusterSet.
 type ClusterSetSpec struct {
+	// ClusterSetID identifies the ClusterSet.
+	// +kubebuilder:validation:Required
+	ClusterSetID string `json:"clusterSetID"`
+	// ClusterID identifies the local cluster.
+	// +kubebuilder:validation:Required
+	ClusterID string `json:"clusterID"`
 	// Members include member clusters known to the leader clusters.
 	// Used in leader cluster.
-	Members []MemberCluster `json:"members,omitempty"`
+	//
+	// DEPRECATED
+	// This field is not used for ClusterSet setup and planned to be removed
+	// in the future releases.
+	Members []LeaderClusterInfo `json:"members,omitempty"`
 	// Leaders include leader clusters known to the member clusters.
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=1
 	// +kubebuilder:validation:Required
-	Leaders []MemberCluster `json:"leaders"`
+	Leaders []LeaderClusterInfo `json:"leaders"`
 	// The leader cluster Namespace in which the ClusterSet is defined.
 	// Used in member cluster.
-	Namespace string `json:"namespace,omitempty"`
+	// +kubebuilder:validation:Required
```

Comment:
If it is not used in leader, it cannot be Required?

Reply:
You are right, I removed this, thanks.